lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3417,7 +3413,28 @@ namespace {
       }
       result->setIsObjC(false);
       result->setIsDynamic(false);
-      result->setInterfaceType(type);
+
+      // If struct field is a weak objc object pointer, create a weak storage type
+      // and add the appropriate wrapper to the param
+      if (objcLifetime == clang::Qualifiers::OCL_Weak) {
+        if (auto nullability =
+              decl->getType()->getNullability(Impl.getClangASTContext()))
+        {
+          if (*nullability == clang::NullabilityKind::NonNull) {
+            Impl.addImportDiagnostic(
```

Comment:
We seen to be missing a test case that exercise this diagnostic.

Reply:
Also is it possible for an objc member to be `weak` and `non-null`? I thought weak was implicit always nullable. e.g. if we try to declare a property `(weak, nonnull)` we get `property attributes 'nonnull' and 'weak' are mutually exclusive`. So this is more a question because, mostly curious on this =]